test/unit-test/cellular_3gpp_api_utest.c

```diff
@@ -0,0 +1,5068 @@
+ */
+void test_Cellular_CommonGetEidrxSettings_Cb_Cellular_RecvFuncGetEidrxSettings_Null_Context( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularHandle_t cellularHandle;
+    CellularEidrxSettingsList_t eidrxSettingsList;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_IsValidPdn_IgnoreAndReturn( CELLULAR_SUCCESS );
+    cbCondition = 0;
+    _Cellular_AtcmdRequestWithCallback_StubWithCallback( Mock_AtcmdRequestWithCallback__Cellular_Cellular_RecvFuncGetEidrxSettings );
+    _Cellular_TranslatePktStatus_ExpectAndReturn( CELLULAR_PKT_STATUS_FAILURE, CELLULAR_INTERNAL_FAILURE );
+
+    cellularStatus = Cellular_CommonGetEidrxSettings( cellularHandle, &eidrxSettingsList );
+    TEST_ASSERT_EQUAL( CELLULAR_INTERNAL_FAILURE, cellularStatus );
+}
+
+/**
+ * @brief Test that null at command failure case in callback function _Cellular_RecvFuncIpAddress
+ * for Cellular_CommonGetEidrxSettings to return CELLULAR_INTERNAL_FAILURE.
+ */
+void test_Cellular_CommonGetEidrxSettings_Cb_Cellular_RecvFuncIpAddress_Null_AtCmd_Failure( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularHandle_t cellularHandle;
+    CellularEidrxSettingsList_t eidrxSettingsList;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_IsValidPdn_IgnoreAndReturn( CELLULAR_SUCCESS );
+    cbCondition = 1;
+    _Cellular_AtcmdRequestWithCallback_StubWithCallback( Mock_AtcmdRequestWithCallback__Cellular_RecvFuncIpAddress );
+    _Cellular_TranslatePktStatus_ExpectAndReturn( CELLULAR_PKT_STATUS_BAD_PARAM, CELLULAR_INTERNAL_FAILURE );
+
+    cellularStatus = Cellular_CommonGetEidrxSettings( cellularHandle, &eidrxSettingsList );
+    TEST_ASSERT_EQUAL( CELLULAR_INTERNAL_FAILURE, cellularStatus );
+}
+
+/**
+ * @brief Test that null data pointer case in callback function _Cellular_RecvFuncIpAddress
+ * for Cellular_CommonGetEidrxSettings to return CELLULAR_INTERNAL_FAILURE.
+ */
+void test_Cellular_CommonGetEidrxSettings_Cb_Cellular_RecvFuncIpAddress_Null_Data_Pointer( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularHandle_t cellularHandle;
+    CellularEidrxSettingsList_t eidrxSettingsList;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_IsValidPdn_IgnoreAndReturn( CELLULAR_SUCCESS );
+    cbCondition = 2;
+    _Cellular_AtcmdRequestWithCallback_StubWithCallback( Mock_AtcmdRequestWithCallback__Cellular_RecvFuncIpAddress );
+    _Cellular_TranslatePktStatus_ExpectAndReturn( CELLULAR_PKT_STATUS_BAD_PARAM, CELLULAR_INTERNAL_FAILURE );
+
+    cellularStatus = Cellular_CommonGetEidrxSettings( cellularHandle, &eidrxSettingsList );
+    TEST_ASSERT_EQUAL( CELLULAR_INTERNAL_FAILURE, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case in callback function _Cellular_RecvFuncIpAddress
+ * for Cellular_CommonGetEidrxSettings to return CELLULAR_INTERNAL_FAILURE.

```

Comment:
Where is this failure code being returned?

Reply:
My bad, the return code should be `CELLULAR_SUCCESS`.